src/clj/fluree/db/query/api.cljc

```diff
@@ -209,36 +208,36 @@
         (dataset db-map defaults)))))
 
 (defn query-connection-fql
-  [conn query did]
+  [conn query request-opts]
   (go-try
-   (let [{:keys [t opts] :as query*} (-> query
-                                         syntax/coerce-query
-                                         (update :opts sanitize-query-options did))
-         default-aliases (some-> query* :from util/sequential)
-         named-aliases   (some-> query* :from-named util/sequential)]
-     (if (or (seq default-aliases)
-             (seq named-aliases))
-       (let [s-ctx    (ctx-util/extract query)
-             ds       (<? (load-dataset conn default-aliases named-aliases t
-                                        s-ctx opts))
-             query**  (update query* :opts dissoc :meta :max-fuel ::util/track-fuel?)
-             max-fuel (:max-fuel opts)]
-         (if (::util/track-fuel? opts)
-           (<? (track-query ds max-fuel query**))
-           (<? (fql/query ds query**))))
-       (throw (ex-info "Missing ledger specification in connection query"
-                       {:status 400, :error :db/invalid-query}))))))
-
+    (let [{:keys [t opts] :as sanitized-query} (-> query
+                                                   syntax/coerce-query
+                                                   (update :opts sanitize-query-options (:did request-opts)))
+          
+          default-aliases (some-> sanitized-query :from util/sequential)
+          named-aliases   (some-> sanitized-query :from-named util/sequential)]
+      (if (or (seq default-aliases)
+              (seq named-aliases))
+        (let [s-ctx    (ctx-util/extract query)
+              ds       (<? (load-dataset conn default-aliases named-aliases t
+                                            s-ctx opts))
+              trimmed-query (update sanitized-query :opts dissoc :meta :max-fuel ::util/track-fuel?)
+              max-fuel (:max-fuel opts)]
+          (if (::util/track-fuel? opts)
+            (<? (track-query ds max-fuel trimmed-query))
+            (<? (fql/query ds trimmed-query))))
+        (throw (ex-info "Missing ledger specification in connection query"
+                        {:status 400, :error :db/invalid-query}))))))
 
 (defn query-connection-sparql
-  [conn query did]
+  [conn query {:keys [role did] :as request-opts}]
   (go-try
-   (let [fql (sparql/->fql query)]
-     (<? (query-connection-fql conn fql did)))))
+    (let [fql              (sparql/->fql query)
+          credentialed-fql (update fql :opts #(merge {:role role :did did} %))]
```

Comment:
Instead of merging the opts into the fql, I think we can just pass them as `opts` to `query-connection-fql`.

Reply:
I don't think the :role makes its way into the query at any point if I just include it as `opts` in `query-connection-fql` unless that isn't how roles are supposed to work anymore?

Reply:
I don't think we use `role` anymore. @bplatz did we deprecate the `role` opt?

Reply:
Yes, `role` is no longer part of policy enforcement nor used anywhere else.